substrate/utils/wasm-builder/src/lib.rs

```diff
@@ -423,36 +392,35 @@ enum RuntimeTarget {
 }
 
 impl RuntimeTarget {
-	fn rustc_target(self) -> &'static str {
-		match self {
-			RuntimeTarget::Wasm => "wasm32-unknown-unknown",
-			RuntimeTarget::Riscv => "riscv32ema-unknown-none-elf",
+	/// Creates a new instance.
+	fn new() -> Option<Self> {
+		let Some(value) = env::var_os(RUNTIME_TARGET) else {
+			return Some(Self::Wasm);
+		};
+		if value == "wasm" {
+			Some(Self::Wasm)
+		} else if value == "riscv" {
+			Some(Self::Riscv)
+		} else {
+			None
 		}
 	}
 
-	fn build_subdirectory(self) -> &'static str {
-		// Keep the build directories separate so that when switching between
-		// the targets we won't trigger unnecessary rebuilds.
-		match self {
-			RuntimeTarget::Wasm => "wbuild",
-			RuntimeTarget::Riscv => "rbuild",
+	/// Figures out the target parameter value for rustc.
+	fn parameter(self) -> String {
+		if self == RuntimeTarget::Wasm {
```

Comment:
Use a `match` instead of `==` to guarantee every variant is exhaustively handled.